### PR TITLE
Fix variable case-sensitivity

### PR DESCRIPTION
**Pull request by Syst3ms**
Target Minecraft versions: any
Requirements: N/A
Related issues: #992 

Description:
Fixes variable case-sensitivity by turning the variable name to lower case in getVariable and setVariable, and removing now redundant toLowerCase calls in Variable mainly.

I'm recreating this PR because I messed up with git.

### CODE REVIEW

**Comment by Syst3ms** [comment]
Well, not like I can do much now, except reverting and recommiting, which is 3 commits for nothing.

**Comment by xXAndrew28Xx** [comment]
I guess it’s fine, or just revert the .idea changes in one commit.

**Comment by Syst3ms** [comment]
Yeah actually I can do that. I'll do that this evening.

**Comment by Syst3ms** [comment]
Done.

**Comment by Syst3ms** [comment]
There, added config option, should be good for merge.

**Comment by bensku** [comment]
@Snow-Pyon Could you test and merge this? I'm a bit busy until next weekend.

**Comment by Snow-Pyon** [comment]
Sure, I'll be merging this if tests go well.

File: src/main/java/ch/njol/skript/ScriptLoader.java [hunk 1]
```diff
@@ -526,7 +526,7 @@ public String run(final Matcher m) {
 								Skript.error("Invalid use of percent signs in variable name");
 								continue;
 							}
-							if (Variables.getVariable(name, null, false) != null)
+							if (Variables.getVariable(n.getKey(), null, false) != null)
```
Comment (Snow-Pyon) [inline src/main/java/ch/njol/skript/ScriptLoader.java]: It doesn't compile due to this change

File: src/main/java/ch/njol/skript/SkriptConfig.java [hunk 2]
```diff
@@ -100,6 +101,8 @@ public void set(final Timespan t) {
 	
 	public final static Option<Boolean> usePlayerUUIDsInVariableNames = new Option<Boolean>("use player UUIDs in variable names", false); // TODO change to true later (as well as in the default config)
 	public final static Option<Boolean> enablePlayerVariableFix = new Option<Boolean>("player variable fix", true);
+
+	public final static Option<Boolean> caseInsensitiveVariables = new Option<Boolean>("case-insensitive variables", false);
```
Comment (Snow-Pyon) [inline src/main/java/ch/njol/skript/SkriptConfig.java]: Should be true by default, would break Skript's default behaviour otherwise (setting `{_SomEthing}` will actually set `{_SomEthing}` and not `{_something}` as intended.

**Comment by Snow-Pyon** [comment]
You also forgot to readd the `profiles_settings.xml` and `encodings.xml` from the idea folder, which doesn't really matter but it'd better to handle that in #1053.

**Comment by Syst3ms** [comment]
I had forgotten about this, just committed the requested changes.

File: src/main/java/ch/njol/skript/ScriptLoader.java [hunk 3]
```diff
@@ -526,7 +526,7 @@ public String run(final Matcher m) {
 								Skript.error("Invalid use of percent signs in variable name");
 								continue;
 							}
-							if (Variables.getVariable(name, null, false) != null)
+							if (Variables.getVariable(((EntryNode) n).getKey(), null, false) != null)
```
Comment (bensku) [inline src/main/java/ch/njol/skript/ScriptLoader.java]: Why this change? Would changing the name variable not be better?

File: src/main/java/ch/njol/skript/variables/Variables.java [hunk 4]
```diff
@@ -272,15 +273,19 @@ static Lock getReadLock() {
 	 */
 	@Nullable
 	public final static Object getVariable(final String name, final @Nullable Event e, final boolean local) {
-		if (local) {
+		String n = name;
+        if (SkriptConfig.caseInsensitiveVariables.value()) {
```
Comment (bensku) [inline src/main/java/ch/njol/skript/variables/Variables.java]: Querying config options can be potentially expensive, so it should not be done in a method that is called very often. Suggestion: use changer in the option to set a static field somewhere, then check against it.

File: src/main/java/ch/njol/skript/variables/Variables.java [hunk 5]
```diff
@@ -294,8 +299,12 @@ public final static Object getVariable(final String name, final @Nullable Event
 	 * @param value The variable's value. Use <tt>null</tt> to delete the variable.
 	 */
 	public final static void setVariable(final String name, @Nullable Object value, final @Nullable Event e, final boolean local) {
-		if (value != null) {
-			assert !name.endsWith("::*");
+        String n = name;
+        if (SkriptConfig.caseInsensitiveVariables.value()) {
```
Comment (bensku) [inline src/main/java/ch/njol/skript/variables/Variables.java]: Do not query config directly there either.

File: src/main/java/ch/njol/skript/variables/Variables.java (hunk 4 above)
Comment (Snow-Pyon) [reply src/main/java/ch/njol/skript/variables/Variables.java]: I don't think that'd be the case as config values are only parsed on startup & `/sk reload config/all` but if you think so, I'll make this change myself.

**Comment by Snow-Pyon** [comment]
Thanks for the contribution!
